components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -310,6 +310,10 @@ public List<UserSession> getSessions(int tenantId, List<ExpressionNode> filter,
 
     private void generateApplicationFromAppID(List<Application> applications) throws SessionManagementServerException {
 
+        if (applications == null || applications.isEmpty()) {
```

Comment:
We should be able to use the following to simplify
if (CollectionUtils.isEmpty(applications)) {
   return;
}

Reply:
Fixed